navigator/backend/src/main/scala/com/digitalasset/navigator/json/ModelCodec.scala

```diff
@@ -99,15 +99,17 @@ object ModelCodec {
       def write(value: Contract): JsValue = JsObject(
         "id" -> value.id.toJson,
         "template" -> value.template.toJson,
-        "argument" -> value.argument.toJson
+        "argument" -> value.argument.toJson,
+        "agreementText" -> value.agreementText.toJson
       )
       def read(value: JsValue, types: DamlLfTypeLookup): Contract = {
         val id = anyField(value, "id", "Contract").convertTo[ApiTypes.ContractId]
         val template = anyField(value, "template", "Contract").convertTo[Template]
         val argument = ApiCodecCompressed
           .jsValueToApiType(anyField(value, "record", "Contract"), template.id, types)
           .asInstanceOf[ApiRecord]
-        Contract(id, template, argument)
+        val agreementText = optionStrField(value, "agreementText", "Contract")
```

Comment:
I think there is an issue here.

The `write` method always includes a `agreementText` property, presumably with a `null` value when the agreement text is missing (I expect `None.toJson` returns `JsNull`).
The `read` method only handles cases where the property is missing, or where its value is a string (see optionStrField). How about handling `case Some(JsNull) => None` in `optionStrField`? 

I realize there is no corresponding test. For other codecs, we have tests, e.g., https://github.com/digital-asset/daml/blob/master/navigator/backend/src/test/scala/com/digitalasset/navigator/backend/json/DamlLfCodecSpec.scala#L28

Reply:
Turns out that this `read` method can be deleted as it isn't used anywhere.

Reply:
Oh good catch. I'm not sure this whole class is needed at all - I can't think of any place where we would be serializing contract objects (the GraphQL API uses a different mechanism for serializing data).

Reply:
See f4dd868 and 96943e4